File: java/reverse-string/src/main/java/ReverseString.java
import java.util.ArrayList;
import java.util.stream.Collectors;

class ReverseString {
    String reverse(String inputString) {
        return inputString.chars()
                .collect(() -> new ArrayList<Integer>(), (acc, n) -> acc.add(0, n), ArrayList::addAll)
                .stream().map(Character::toString).collect(Collectors.joining());
    }
}

File: c/reverse-string/reverse_string.c

#include "reverse_string.h"

char *reverse(const char *s) {
    int len = (int) strlen(s);
    char *res = calloc(len, sizeof(char));
    for (int i = 0; i < len; ++i) {
        res[len - i - 1] = s[i];
    }
    return res;
}
